src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php

```diff
@@ -85,8 +85,10 @@ public function build(ContainerBuilder $container)
         $container->addCompilerPass(new AddCacheWarmerPass());
         $container->addCompilerPass(new AddCacheClearerPass());
         $container->addCompilerPass(new AddExpressionLanguageProvidersPass());
-        $container->addCompilerPass(new TranslationExtractorPass());
-        $container->addCompilerPass(new TranslationDumperPass());
+        if (class_exists('Symfony\Component\Translation\Translator')) {
```

Comment:
all other compiler passes are always registered, and return early when the service does not exist (accounting for disabling by configuration too). Shouldn't it be done here too ?